mpz-core/src/ggm_tree.rs

```diff
@@ -18,15 +18,15 @@ impl GgmTree {
         Self { tkprp, depth }
     }
 
-    /// Input: `seed`: a seed.
-    /// Output: `tree`: a GGM (binary tree) `tree`, with size `2^{depth-1}`
-    /// Output: `k0`: XORs of all the left-node values in each level, with size `depth-1`.
-    /// Output: `k1`: XORs of all the right-node values in each level, with size `depth-1`.
+    /// Input : `seed` - a seed.
+    /// Output: `tree` - a GGM (binary tree) `tree`, with size `2^{depth}`.
+    /// Output: `k0` - XORs of all the left-node values in each level, with size `depth`.
+    /// Output: `k1`- XORs of all the right-node values in each level, with size `depth`.
     /// This implementation is adapted from EMP Toolkit.
     pub fn gen(&self, seed: Block, tree: &mut [Block], k0: &mut [Block], k1: &mut [Block]) {
-        assert!(tree.len() == 1 << (self.depth - 1));
-        assert!(k0.len() == self.depth - 1);
-        assert!(k1.len() == self.depth - 1);
+        assert!(tree.len() == 1 << (self.depth));
+        assert!(k0.len() == self.depth);
+        assert!(k1.len() == self.depth);
         let mut buf = vec![Block::ZERO; 8];
```

Comment:
I think we can avoid this heap allocation

```suggestion
        let mut buf = [Block::ZERO; 8];
```